### PR TITLE
Migrate `getWeekOfMonth` to typescript

### PR DESCRIPTION
**Pull request by codecog**
This change moves `getWeekOfMonth` over to typescript. A few key points:
- Removed test to assert that the method will throw if the required amount of arguments are not passed as this can be checked with types
- added `@babel/plugin-proposal-nullish-coalescing-operator` to take advantage of setting a default to `weekStartsOn` of `0` by using `??`
- Updated parts of the JSDoc comments to use tsdoc

### CODE REVIEW

**Comment by codecog** [comment]
@kossnocorp I had issues running `./scripts/build/build.sh` as it throws an error to do with the docs. Have you run into this problem before?

File: src/getWeekOfMonth/test.js [hunk 1]
```diff
@@ -71,10 +71,6 @@ describe('getWeekOfMonth', function() {
     assert(isNaN(result))
   })
 
-  it('throws TypeError exception if passed less than 1 argument', function() {
-    assert.throws(getWeekOfMonth.bind(null), TypeError)
-  })
-
```
Comment (kossnocorp) [inline src/getWeekOfMonth/test.js:77]: Please keep this and the check in the code and also don't forget to keep `@throws` in the docs. I know that TypeScript solves the problem and we'll remove all type checks in v3, but this is a breaking change and we're not ready for a major release.

File: src/getWeekOfMonth/index.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    startWeekDay >= weekStartsOn
+      ? weekStartsOn + 7 - startWeekDay
+      : weekStartsOn - startWeekDay
+
+  return dayOfMonth > lastDayOfFirstWeek
+    ? weekNumber + Math.ceil((dayOfMonth - lastDayOfFirstWeek) / 7)
+    : weekNumber
+}
+
+export default getWeekOfMonth
```
Comment (kossnocorp) [inline src/getWeekOfMonth/index.ts:55]: Please rewrite to a normal function `export default function getWeekOfMonth`.

File: src/getWeekOfMonth/index.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * @param date - The given date
+ * @param options - An object with options
+ * @param [options.locale=defaultLocale] - The locale object. See {@link https://date-fns.org/docs/Locale Locale}
+ * @param [options.weekStartsOn=0] - The index of the first day of the week (0 - Sunday)
+ * @returns The week of month
+ *
+ * @example // Which week of the month is 9 November 2017?
+ * const result = getWeekOfMonth(new Date(2017, 10, 9));
+ */
```
Comment (kossnocorp) [inline src/getWeekOfMonth/index.ts:30]: Please preserve the original JSDoc, until v3 when all code is in TypeScript we're forced to use it.

File: src/getWeekOfMonth/index.js [hunk 4]
```diff
@@ -1,72 +0,0 @@
...
-  }
-
-  var weekNumber = 1
-
-  if (currentDayOfMonth > lastDayOfFirstWeek) {
-    var remainingDaysAfterFirstWeek = currentDayOfMonth - lastDayOfFirstWeek
-    weekNumber = weekNumber + Math.ceil(remainingDaysAfterFirstWeek / 7)
-  }
-  return weekNumber
-}
```
Comment (kossnocorp) [inline src/getWeekOfMonth/index.js:72]: Could you please rework your commit so git understands that it is a rename to preserve the git history?

File: src/getWeekOfMonth/index.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+const getWeekOfMonth = (
+  date: Date | number,
+  { locale, weekStartsOn } = defaultOptions
```
Comment (kossnocorp) [inline src/getWeekOfMonth/index.ts:33]: Please assign type `LocaleOptions & WeekStartOptions` from https://github.com/date-fns/date-fns/blob/master/src/types.ts. Also `= defaultOptions` breaks when you pass any object, so please get rid of it and keep it in the code or `{ locale, weekStartsOn = 0 }`.

**Review by kossnocorp — CHANGES_REQUESTED** [review]
Thank you for the pull request! Please address my comments. Docs generation will fix once you restore the original JSDoc. Also, please rebase from the master branch, where we fixed tests.

**Pull request closed without merging**